src/main/java/htsjdk/samtools/SAMRecordUtil.java

```diff
@@ -26,56 +26,96 @@
-            SAMTagUtil.getSingleton().OQ
-    };
+    private static List<String> TAGS_TO_REVERSE_COMPLEMENT = Arrays.asList(SAMTag.E2.name(), SAMTag.SQ.name());
+    private static List<String> TAGS_TO_REVERSE            = Arrays.asList(SAMTag.OQ.name(), SAMTag.U2.name());
 
     /**
      * Reverse-complement all known sequence and base quality attributes of the SAMRecord.
      */
     public static void reverseComplement(final SAMRecord rec) {
+        reverseComplement(rec, TAGS_TO_REVERSE_COMPLEMENT, TAGS_TO_REVERSE);
+    }
+
+    public static void reverseComplement(final SAMRecord rec, Collection<String> tagsToRevcomp, Collection<String> tagsToReverse) {
         final byte[] readBases = rec.getReadBases();
         SequenceUtil.reverseComplement(readBases);
         rec.setReadBases(readBases);
         final byte qualities[] = rec.getBaseQualities();
         reverseArray(qualities);
         rec.setBaseQualities(qualities);
-        final byte[] sqTagValue = (byte[])rec.getAttribute(SAMTagUtil.getSingleton().SQ);
-        if (sqTagValue != null) {
-            SQTagUtil.reverseComplementSqArray(sqTagValue);
-            rec.setAttribute(SAMTagUtil.getSingleton().SQ, sqTagValue);
-        }
-        final String e2TagValue = (String)rec.getAttribute(SAMTagUtil.getSingleton().E2);
-        if (e2TagValue != null) {
-            final byte[] secondaryBases = StringUtil.stringToBytes(e2TagValue);
-            SequenceUtil.reverseComplement(secondaryBases);
-            rec.setAttribute(SAMTagUtil.getSingleton().E2, StringUtil.bytesToString(secondaryBases));
+
+        // Deal with tags that need to be reverse complemented
+        if (tagsToRevcomp != null) {
+            for (final String tag: tagsToRevcomp) {
+                Object value = rec.getAttribute(tag);
+                if (value != null) {
+                    if (value instanceof byte[]) SequenceUtil.reverseComplement((byte[]) value);
+                    else if (value instanceof String) value = SequenceUtil.reverseComplement((String) value);
+                    else throw new IllegalArgumentException("Don't know how to reverse complement: " + value);
+                    rec.setAttribute(tag, value);
+                }
+            }
         }
-        for (final short stringTag : STRING_TAGS_TO_REVERSE) {
-            final String value = (String)rec.getAttribute(stringTag);
-            if (value != null) {
-                rec.setAttribute(stringTag, StringUtil.reverseString(value));
+
+        // Deal with tags that needed to just be reversed
+        if (tagsToReverse != null) {
+            for (final String tag : tagsToReverse) {
+                Object value = rec.getAttribute(tag);
+                if (value != null) {
+                    if (value instanceof String) {
+                        value = StringUtil.reverseString((String) value);
+                    }
+                    else if (value.getClass().isArray()) {
```

Comment:
Just out of curiosity, why not maintain the previous behavior (converting to String and then reversing that)?  Code was a lot cleaner.  Is it b/c of bad performance?
Maybe pull all this out into a reverseArray() method that takes in an Object for code cleanliness?